main.rb

```diff
@@ -2,113 +2,194 @@
 def add_business_day_hours(start_time, hours)
+  return start_time if hours <= 0
+  
   time = start_time.dup
   
   hours.times do
-    time += 3600  # Add one hour
-    # Skip weekends
-    time += 1 * 24 * 3600 if time.wday == 0  # Skip Sunday -> Monday
-    time += 2 * 24 * 3600 if time.wday == 6  # Skip Saturday -> Sunday -> Monday
+    time += SECONDS_IN_HOUR
+    # Skip weekends: if we land on Sunday (0), move to Monday
+    time += 1 * HOURS_IN_DAY * SECONDS_IN_HOUR if time.utc.wday == 0
+    # Skip weekends: if we land on Saturday (6), move to Monday
+    time += 2 * HOURS_IN_DAY * SECONDS_IN_HOUR if time.utc.wday == 6
   end
   
   time
 end
 
-GITHUB_TOKEN = ENV['GITHUB_TOKEN']
-REMINDER_MESSAGE = ENV['REMINDER_MESSAGE']
-REVIEW_TURNAROUND_HOURS = ENV['REVIEW_TURNAROUND_HOURS']
-PROCESS_REMINDER_MESSAGE = ENV['PROCESS_REMINDER_MESSAGE']
-PROCESS_REVIEW_TURNAROUND_HOURS = ENV['PROCESS_REVIEW_TURNAROUND_HOURS']
-REPO = ENV['GITHUB_REPOSITORY']
+# Check if a specific comment already exists in the comments list
+def comment_exists?(comments, target_comment)
+  return false if comments.nil? || target_comment.nil? || target_comment.empty?
+  comments.any? { |comment| comment[:body]&.include?(target_comment) }
+end
 
-client = Octokit::Client.new(access_token: GITHUB_TOKEN, per_page: 100)
+# Create and post a comment to GitHub
+def create_comment(client, repo, pr_number, comment_text, comment_type = "comment")
+  return false if comment_text.nil? || comment_text.empty?
+  
+  client.add_comment(repo, pr_number, comment_text)
+  puts "#{comment_type.capitalize} created: #{comment_text}"
+  true
+rescue => e
+  puts "Error creating #{comment_type}: #{e.message}"
+  false
+end
 
-begin
-  pull_requests = client.pull_requests(REPO, state: 'open')
+# Check if reviewer was re-requested for review after their last changes request
+def reviewer_re_requested_after_changes?(timeline, reviewer_login, changes_requested_at)
+  return false if timeline.nil? || reviewer_login.nil? || changes_requested_at.nil?
+  
+  timeline.any? do |event|
+    event[:event] == 'review_requested' &&
+    event[:requested_reviewer] &&
+    event[:requested_reviewer][:login] == reviewer_login &&
+    event[:created_at] > changes_requested_at
+  end
+end
+
+validate_environment!
+client = Octokit::Client.new(access_token: GITHUB_TOKEN, per_page: API_PER_PAGE)
+
+# Send reviewer reminders for overdue reviews
+def send_reviewer_reminders(client, repo, pr, timeline, comments, current_time)
+  review_requested_events = timeline.select { |e| e[:event] == 'review_requested' }
+  requested_reviewers = pr[:requested_reviewers]&.map { |rr| rr[:login] } || []
+
+  if review_requested_events.empty? || requested_reviewers.empty?
+    puts "  No pending reviews for PR ##{pr[:number]}."
+    return
+  end
+
+  # Calculate deadline based on most recent review request (sorted by timestamp)
+  last_review_request_time = review_requested_events.max_by { |e| e[:created_at] }[:created_at]
```

Comment:
What was wrong with `review_requested_events.last[:created_at]`? That seems simpler.